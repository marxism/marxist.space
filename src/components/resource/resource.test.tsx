/* eslint-env jest */

import * as React from 'react'
import Resource from '.'
import { create } from 'react-test-renderer'

describe('resource', () => {
  test('renders with minimum props', () => {
    const props = {
      href: 'https://marxist.space',
      title: 'marxist.space',
      tags: ['links', 'collection'],
    }
    expect(create(<Resource {...props} />).toJSON()).toMatchSnapshot()
  })

  test('renders with all props', () => {
    const props = {
      href: 'https://marxist.space',
      title: 'marxist.space',
      tags: ['links', 'collection'],
      description: 'Links for you!',
      excerpts: ['So many good links', 'The best links'],
    }
    expect(create(<Resource {...props} />).toJSON()).toMatchSnapshot()
  })
})
