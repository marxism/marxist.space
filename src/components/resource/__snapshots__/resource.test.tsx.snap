// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resource renders with all props 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 16px;
  line-height: 21px;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.1s ease;
  transition: color 0.1s ease;
  text-underline-position: under;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 small {
  font-size: 12px;
}

.c2 {
  max-width: 800px;
  padding-bottom: 1px;
  padding-left: 8px;
  font-size: 13px;
  line-height: 19px;
}

.c3 {
  font-size: 12px;
  margin-top: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-left: 1px solid;
  padding: 8px;
  min-width: 0;
  margin-bottom: 8px;
}

<div
  className="c0"
>
  <a
    className="c1"
    href="https://marxist.space"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    marxist.space
    <small>
      Links for you!
    </small>
  </a>
  <div
    className="c2"
  >
    So many good links
  </div>
  <div
    className="c2"
  >
    The best links
  </div>
  <span
    className="c3"
  >
    Tags: 
    links collection
  </span>
</div>
`;

exports[`resource renders with minimum props 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 16px;
  line-height: 21px;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.1s ease;
  transition: color 0.1s ease;
  text-underline-position: under;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 small {
  font-size: 12px;
}

.c2 {
  font-size: 12px;
  margin-top: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-left: 1px solid;
  padding: 8px;
  min-width: 0;
  margin-bottom: 8px;
}

<div
  className="c0"
>
  <a
    className="c1"
    href="https://marxist.space"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    marxist.space
  </a>
  <span
    className="c2"
  >
    Tags: 
    links collection
  </span>
</div>
`;
